pandas/tests/indexing/test_indexing.py

```diff
@@ -58,6 +60,93 @@ def test_setitem_ndarray_1d(self):
         with pytest.raises(ValueError):
             df[2:5] = np.arange(1, 4) * 1j
 
+    @pytest.mark.parametrize('index', tm.all_index_generator(5),
+                             ids=lambda x: type(x).__name__)
+    @pytest.mark.parametrize('obj', [
+        lambda i: Series(np.arange(len(i)), index=i),
+        lambda i: DataFrame(
+            np.random.randn(len(i), len(i)), index=i, columns=i)
+    ], ids=['Series', 'DataFrame'])
+    @pytest.mark.parametrize('idxr, idxr_id', [
```

Comment:
might be worth making these a fixture at somepoint (the iteration over indexers)

Reply:
> the iteration over indexers

yeah. i appreciate we have duplication of parameterization at the moment, and fixtures may be of use. didn't carry on down that route at the moment as i wasn't sure if these tests were what was required.

separately for iteration over the indexes:

xref https://github.com/pandas-dev/pandas/pull/25748#discussion_r266411287. There is a `indices` fixture in `pandas\tests\indexes\conftest.py`. Maybe worth promoting that up a level so that it can also be used in `pandas\tests\indexing`